localstack/services/secretsmanager/provider.py

```diff
@@ -68,143 +84,458 @@ def __init__(self):
+    #
+    #  - The AWSPENDING and AWSCURRENT staging labels are attached to the
+    #    same version of the secret, or
+    #  - The AWSPENDING staging label is not attached to any version of the secret.
+    #
+    # If the AWSPENDING staging label is present but not attached to the same
+    # version as AWSCURRENT then any later invocation of RotateSecret assumes
+    # that a previous rotation request is still in progress and returns an error.
+    try:
+        version = next(
+            version
+            for version in secret.versions.values()
+            if AWSPENDING in version["version_stages"]
+        )
+        if AWSCURRENT in version["version_stages"]:
+            msg = "Previous rotation request is still in progress."
+            raise InvalidRequestException(msg)
+
+    except StopIteration:
+        # Pending is not present in any version
+        pass
+
+    if client_request_token:
+        self._client_request_token_validator(client_request_token)
+        new_version_id = client_request_token
+    else:
+        new_version_id = str(uuid.uuid4())
+
+    # We add the new secret version as "pending". The previous version remains
+    # as "current" for now. Once we've passed the new secret through the lambda
+    # rotation function (if provided) we can then update the status to "current".
+    self._add_secret(
+        secret_id,
+        None,
+        description=secret.description,
+        tags=secret.tags,
+        version_id=new_version_id,
+        version_stages=[AWSPENDING],
+    )
+    secret.rotation_lambda_arn = rotation_lambda_arn or ""
+    if rotation_rules:
+        secret.auto_rotate_after_days = rotation_rules.get(rotation_days, 0)
+    if secret.auto_rotate_after_days > 0:
+        secret.rotation_enabled = True
+
+    # Begin the rotation process for the given secret by invoking the lambda function.
+    if secret.rotation_lambda_arn:
+        from moto.awslambda.models import lambda_backends
+
+        lambda_backend = lambda_backends[self.region]
+
+        request_headers = {}
+        response_headers = {}
+
+        func = lambda_backend.get_function(secret.rotation_lambda_arn)
+        if not func:
+            msg = "Resource not found for ARN '{}'.".format(secret.rotation_lambda_arn)
+            raise ResourceNotFoundException(msg)
+
+        for step in ["create", "set", "test", "finish"]:
+            func.invoke(
+                json.dumps(
+                    {
+                        "Step": step + "Secret",
+                        "SecretId": secret.name,
+                        "ClientRequestToken": new_version_id,
+                    }
+                ),
+                request_headers,
+                response_headers,
+            )
+
+        secret.set_default_version_id(new_version_id)
+        version_stages = secret.versions[new_version_id]["version_stages"]
+        if AWSPENDING in version_stages:
+            version_stages.remove(AWSPENDING)
```

Comment:
I feel like this section is similar to above where it's going beyond what AWS does, which can lead to unintended behavior.  While it doesn't impact me, it may be worth considering removing this so localstack can better mirror AWS behavior.

Reply:
This seems to remove AWSPENDING from the now AWSCURRENT secret version once the rotation has terminated.